apps/angular/projection/src/app/app.facade.ts

```diff
@@ -0,0 +1,84 @@
+import { OnDestroy } from '@angular/core';
+import { Injectable } from '@angular/core';
+import { takeUntil } from 'rxjs';
+import { Subject } from 'rxjs';
+import { Observable } from 'rxjs';
+import { randomCity } from './data-access/fake-http.service';
+import { randStudent } from './data-access/fake-http.service';
+import { randTeacher } from './data-access/fake-http.service';
+import { FakeHttpService } from './data-access/fake-http.service';
+import { TeacherStore } from './data-access/teacher.store';
+import { StudentStore } from './data-access/student.store';
+import { CityStore } from './data-access/city.store';
+import { Teacher } from './model/teacher.model';
+import { Student } from './model/student.model';
+import { City } from './model/city.model';
+
+@Injectable()
+export class AppFacade implements OnDestroy {
+  readonly teachers$: Observable<Teacher[]> = this.teacherStore.teachers$;
+  readonly students$: Observable<Student[]> = this.studentStore.students$;
+  readonly cities$: Observable<City[]> = this.cityStore.cities$;
```

Comment:
I don't really understand why you put everything inside the `AppComponent`. The component is handling a lot of logic which should be handle by their own components. 

What will happen if you add one of two more cards, this component will become bigger and bigger. 

What do syou think ?

Reply:
Yes because this is a very small application to only work on specific use case. But I get your point, however I'm not very fan of this big app component. I prefer to split the logic per component since each card correspond to a different business domain. So each card item is a smart computer, the `CardComponent` is the dumb component.

Reply:
I see, thank you for your feedback!